rotkehlchen/chain/evm/decoding/curve_lend/balances.py

```diff
@@ -0,0 +1,164 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.balance import Balance, BalanceSheet
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.ethereum.interfaces.balances import BalancesSheetType, ProtocolWithBalance
+from rotkehlchen.chain.ethereum.utils import token_normalized_value
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.chain.evm.decoding.curve.constants import CPT_CURVE
+from rotkehlchen.chain.evm.decoding.curve_lend.constants import CURVE_VAULT_CONTROLLER_ABI
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.decoder import EVMTransactionDecoder
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class CurveLendBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            tx_decoder: 'EVMTransactionDecoder',
+    ):
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            tx_decoder=tx_decoder,
+            counterparty=CPT_CURVE,
+            deposit_event_types=set(),
+        )
+
+    def _get_controllers_with_balances(self) -> dict[ChecksumEvmAddress, set[ChecksumEvmAddress]]:
+        """Get addresses of controllers for vaults that the user may have balances on.
+        Returns a dict of controller addresses -> set of user addresses with possible balances.
+        """
+        controllers: dict[ChecksumEvmAddress, set[ChecksumEvmAddress]] = defaultdict(set)
+        for address, events in self.addresses_with_activity(event_types={
+            (HistoryEventType.WITHDRAWAL, HistoryEventSubType.GENERATE_DEBT),
+            (HistoryEventType.DEPOSIT, HistoryEventSubType.DEPOSIT_ASSET),
+        }).items():
+            for event in events:
+                if event.extra_data is None or 'vault_controller' not in event.extra_data:
```

Comment:
Actually it is possible for the DEPOSIT/DEPOSIT_ASSET events to not have extra data here when there are deposits to a lending vault (see [here](https://github.com/rotki/rotki/blob/6bc3b6d20f6a929d727ba9581b24edc7218219fa/rotkehlchen/chain/evm/decoding/curve_lend/decoder.py#L166)). These simply need to be skipped without any error, since they do not need processed for balances, so I think it is correct as it is.